test/xds/xds_client_ignores_resource_deletion_test.go

```diff
@@ -0,0 +1,432 @@
+		}
+	}
+}
+
+func testResourceDeletionNotIgnored(t *testing.T, initialResource e2e.UpdateOptions, u func(r *e2e.UpdateOptions)) {
+	t.Helper()
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	t.Cleanup(cancel)
+
+	cc := simulateResourceDeletionOnClient(t, initialResource, u, false)
+	client := testpb.NewTestServiceClient(cc)
+
+	// Spin up go routines to verify RPCs fail after the update.
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+	go func() {
+		for ctx.Err() == nil {
+			if _, err := client.EmptyCall(ctx, &testpb.Empty{}); err != nil {
+				wg.Done()
+				break
+			}
+			time.Sleep(50 * time.Millisecond)
+		}
+	}()
+	go func() {
+		for ctx.Err() == nil {
+			if _, err := client.UnaryCall(ctx, &testpb.SimpleRequest{}); err != nil {
+				wg.Done()
+				break
+			}
+			time.Sleep(50 * time.Millisecond)
+		}
+	}()
+
+	wg.Wait()
+	if ctx.Err() != nil {
+		t.Fatal("Context expired before RPCs failed.")
+	}
+}
+
+func simulateResourceDeletionOnClient(t *testing.T, resources e2e.UpdateOptions, updateResource func(*e2e.UpdateOptions), ignoreResourceDeletion bool) *grpc.ClientConn {
+	t.Helper()
+	ms := startManagementServer(t)
+	bootstrapContent := generateBootstrapContents(t, ms.Address, ignoreResourceDeletion)
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	t.Cleanup(cancel)
+
+	if err := ms.Update(ctx, resources); err != nil {
+		t.Fatal(err)
+	}
+
+	cResolver := generateCustomResolver(t, bootstrapContent)
+	cc, err := grpc.Dial(fmt.Sprintf("xds:///%s", serviceName), grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(cResolver))
+	if err != nil {
+		t.Fatalf("failed to dial local test server: %v", err)
+	}
+	t.Cleanup(func() { cc.Close() })
+
+	makeRPCtoAllEndpoints(t, cc)
+
+	updateResource(&resources)
+	if err := ms.Update(ctx, resources); err != nil {
+		t.Fatal(err)
+	}
+	return cc
+}
+
+func startManagementServer(t *testing.T) *e2e.ManagementServer {
+	ms, err := e2e.StartManagementServer(e2e.ManagementServerOptions{})
+	if err != nil {
+		t.Fatalf("Failed to start management server: %v", err)
+	}
+	t.Cleanup(ms.Stop)
+	return ms
+}
+
+func generateBootstrapContents(t *testing.T, serverURI string, ignoreResourceDeletion bool) []byte {
+	bootstrapContents, err := bootstrap.Contents(bootstrap.Options{
+		NodeID:                             nodeID,
+		ServerURI:                          serverURI,
+		ServerListenerResourceNameTemplate: e2e.ServerListenerResourceNameTemplate,
+		IgnoreResourceDeletion:             ignoreResourceDeletion,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	return bootstrapContents
+}
+
+func generateCustomResolver(t *testing.T, bootstrapContents []byte) resolver.Builder {
+	cResolver, err := resolverBuilder(bootstrapContents)
+	if err != nil {
+		t.Fatalf("Creating xDS resolver for testing: %v", err)
+	}
+	return cResolver
+}
+
+func genericXDSResourceUpdateWithTwoResources(port1 uint32, port2 uint32) e2e.UpdateOptions {
```

Comment:
A lot is happening in this function. Only the endpoints resource is controlled by the parameters. Can we instead define the route config and cluster resources as variables in the only test, `TestIgnoreResourceDeletionOnClient`, that calls this function?

So, in that case, `TestIgnoreResourceDeletionOnClient` would define the route and cluster resources upfront as variables, and the test table would define all an `e2e.UpdateOptions` with all four resources making it more clear what is happening.

You could even define the route and cluster resources with descriptive variable names like `defaultRouteConfigWithTwoRoutes` and `defaultClusterWithTwoEDSServices` or something like that.